pkg/protocol/response.go

```diff
@@ -314,13 +328,21 @@ func (resp *Response) BodyStream() io.Reader {
 //
 // It is safe re-using p after the function returns.
 func (resp *Response) AppendBody(p []byte) {
-	resp.CloseBodyStream()     //nolint:errcheck
+	resp.CloseBodyStream() //nolint:errcheck
```

Comment:
Does `SetBody` also need to add this judge?

Reply:
Extended following discussion conclusions.